apps/studio/src/features/editing-experience/components/EditPageDrawer.tsx

```diff
@@ -1,23 +1,40 @@
+import {
+  type IsomerComponent,
+  type IsomerNativeComponentProps,
+} from '@opengovsg/isomer-components'
 import { useEditorDrawerContext } from '~/contexts/EditorDrawerContext'
 import ComponentSelector from '~/components/PageEditor/ComponentSelector'
 import RootStateDrawer from './RootStateDrawer'
 import TipTapComponent from './TipTapComponent'
 import ComplexEditorStateDrawer from './ComplexEditorStateDrawer'
 
+function narrowToTiptap(
+  component: IsomerComponent,
+): IsomerNativeComponentProps {
+  // TODO: get rid of the cast here
+  return component as IsomerNativeComponentProps
+}
```

Comment:
in root state drawer, when we click into a component, we only select `tiptap` if the component is prose. this means that tiptap will always be able handle the content but because our type of `IsomerComponent` is wider, we have to either validate or cast here.